subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Device.kt

```diff
@@ -82,19 +83,30 @@ object Device {
     }
 
     fun waitForLauncher(timeout: Long = UITestConfig.activityLaunchTimeoutMilliseconds) {
```

Comment:
In app with targetSdk 30 in Android 11 it was broken - always timeout.

The reason is in incorrect `launcherPackageName`:
```kotlin
public String getLauncherPackageName() {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.addCategory(Intent.CATEGORY_HOME);
        PackageManager pm = mInstrumentation.getContext().getPackageManager();
        ResolveInfo resolveInfo = pm.resolveActivity(intent, PackageManager.MATCH_DEFAULT_ONLY); <---
        return resolveInfo.activityInfo.packageName;
}
```
It returns fallback intent to settings, but the real launcher is not accessible:

```
system_process I/AppsFilter: interaction: PackageSetting{9b6897d com.avito.android..../10121} -> PackageSetting{44ad072 com.android.launcher3/10109} BLOCKED
```

I've commented why other options are worse for a target application.